iree/samples/dynamic_shapes/README.md

```diff
@@ -0,0 +1,101 @@
+# "Dynamic Shapes" sample
+
+This sample shows how to
+
+1. Create a TensorFlow program that includes dynamic shapes in program inputs
+   and outputs
+2. Import that program into IREE's compiler
+3. Compile that program to an IREE VM bytecode module
+4. Load the compiled program using IREE's high level runtime C API
+5. Call exported functions on the loaded program
+
+Steps 1-2 are performed in Python via the
+[`dynamic_shapes.ipynb`](./dynamic_shapes.ipynb)
+[Colab](https://research.google.com/colaboratory/) notebook:
+
+[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/google/iree/blob/main/iree/samples/dynamic_shapes/dynamic_shapes.ipynb)
+
+Step 3 should be performed on your development host machine
+
+Steps 4-5 are in [`main.c`](./main.c)
+
+The program used to demonstrate includes functions with varying uses of
```

Comment:
The iota example doesn't really show this well as it's taking a static "scalar tensor" and those are just weird. You could make reduce_sum reduce across 2 dimensions so your input was `[None, 4]` or something to show how you can be partially dynamic.

Reply:
For `iota`, I wanted something taking a static input and producing a dynamic output (to show on the C API side how to handle that case, though it isn't really that different from any of the other cases).

Reply:
Added `reduce_sum_2d` to go along with the `reduce_sum_1d`. How's this look now?